packages/go_router/lib/src/route_data.dart

```diff
@@ -112,8 +112,8 @@ abstract class GoRouteData extends RouteData {
     FutureOr<String?> redirect(BuildContext context, GoRouterState state) =>
         factoryImpl(state).redirect(context, state);
 
-    FutureOr<bool> onExit(BuildContext context, GoRouterState state) =>
-        factoryImpl(state).onExit(context, state);
+    FutureOr<bool>? onExit(BuildContext context, GoRouterState state) =>
+        factoryImpl(state).onExit?.call(context, state);
```

Comment:
instead of doing this, should this pass null to GoRoute if `onExit` is null, and use `!` here

Reply:
this way you also don't need to change the function signuature of ExitCallback

Reply:
Done!